misc/python/materialize/docker.py

```diff
@@ -0,0 +1,83 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""Docker utilities."""
+
+
+from materialize import buildkite, git
+
+try:
+    from semver.version import Version
+except ImportError:
+    from semver import VersionInfo as Version  # type: ignore
+
+
+def resolve_ancestor_image_tag_for_comparison() -> str:
```

Comment:
Using "comparison" in the name of this method would be a misnomer, as we would want the as much of the same logic to apply for both upgrades and comparison-based benchmarks.

Reply:
Will adjust it!